phoenix-core/src/main/java/org/apache/phoenix/expression/BaseJSONExpression.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.phoenix.expression;
+
+import java.util.List;
+
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.phoenix.expression.visitor.ExpressionVisitor;
+import org.apache.phoenix.schema.tuple.Tuple;
+import org.apache.phoenix.schema.types.PDataType;
+
+public abstract class BaseJSONExpression extends BaseCompoundExpression {
+	public BaseJSONExpression(List<Expression> children) {
+        super(children);
+    }
+	public BaseJSONExpression() {
+    }
+	public abstract boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr);
+	public abstract <T> T accept(ExpressionVisitor<T> visitor);
+	public abstract PDataType getDataType();
+	public PDataType getRealDataType(){
+		return null;
```

Comment:
I spoke to @JamesRTaylor  about this and he suggested that the user would have to explicitly cast the expression, (since we don't know what the datatype is at compile time). 
and have getDataType() return PVarbinary . 

CAST ( JSON_COL->'KEY' AS INTEGER )